src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -747,10 +867,13 @@ S2ADecryptStatus s2a_decrypt_record(s2a_crypter* crypter, iovec& record_header,
       /** Check whether the plaintext is a key update message. **/
       if (*plaintext_bytes_written == 5 &&
           memcmp(plaintext, "\x18\x00\x00\x01", 4) == 0 && plaintext[4] < 2) {
-        // TODO(mattstev): update keys. This can only be added once the traffic
-        // secret is stored. To do so, we require that the HKDF_expand function
-        // be added.
-        return S2ADecryptStatus::UNIMPLEMENTED;
+        grpc_status_code key_update_status =
```

Comment:
If I remember correctly, after updating the key we should send an ACK message, right? Is this covered somewhere else in the code?

Reply:
Do you know where this is described? It doesn't seem to be mentioned in the [key update section of the TLS 1.3 RFC](https://tools.ietf.org/html/rfc8446#section-4.6.3), and SSLCryptor doesn't seem to do this.

Reply:
I got the terminology wrong but there is a case where the recipient of a `KeyUpdate` message MUST send a response back. From the same RFC section that you pointed to

> If the request_update field is set to "update_requested", then the
> receiver MUST send a KeyUpdate of its own with request_update set to
> "update_not_requested" prior to sending its next Application Data
> record.

Do you know how is this handled by the other code?